### PR TITLE
add gortsplib sps tests

### PR DESCRIPTION
**Pull request by Curid**
This is all the [gortsplib](https://github.com/bluenviron/gortsplib/blob/862b1061f4a208d172cd8d9ed6c61e29f35541ef/pkg/codecs/h264/sps_test.go) sps tests if you want them.

### CODE REVIEW

**Comment by dholroyd** [comment]
I think apart from the tests, the only non-formatting change is the addition of the `fps()` method, right?  (In future consider putting the rustfmt changes in a separate commit for easier review - and yeah I confess I should have run rustfmt myself - and enforce it in CI!)

**Comment by Curid** [comment]
Sorry about the formatting. I can make two separate PRs for `cargo fmt` and `cargo clippy` first if you want.

gortsplib has a `fps()` method and tests for it. I don't think I have a use for it personally, would you like me to remove it?

**Comment by dholroyd** [comment]
I am happy to keep `fps()` in this PR.

File: src/nal/sps.rs [hunk 1]
```diff
@@ -1090,4 +1094,532 @@ mod test {
...
+                    ..BitstreamRestrictions::default()
+                }),
+                ..VuiParameters::default()
+            }),
+        },
+        1920,
+        1080,
+        30.0; "1920x1080 nvidia"
+    )]
+    // This fails.
```
Comment (dholroyd) [inline src/nal/sps.rs:1287]: Does this failure case represent a bug that needs fixing?  (Be that in this PR or a separate change.)

Comment (Curid) [reply src/nal/sps.rs:1287]: I'm not sure if this case was from a real sample or just used for higher test coverage.

This is the error:
```
`ReaderErrorFor("finish", Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })
```

Removing the comment on the last byte changes the error to this:
```
assertion failed: `(left == right)`
  left: `1084`,
  right: `1080`', h264-reader/src/nal/sps.rs:1619:9
```

**Pull request merged**